static/app/views/issueDetails/streamline/foldSection.spec.tsx

```diff
@@ -0,0 +1,574 @@
+          }),
+        }
+      );
+
+      expect(mockUseIssueDetails.dispatch).not.toHaveBeenCalled();
+    });
+
+    it('uses correct initialCollapse value when preventCollapse is true', () => {
+      render(
+        <FoldSection
+          title="Test Section"
+          sectionKey={SectionKey.HIGHLIGHTS}
+          initialCollapse
+          preventCollapse
+        >
+          <div>Test Content</div>
+        </FoldSection>,
+        {
+          wrapper: makeWrapper({
+            organization: OrganizationFixture({streamlineOnly: false}),
+          }),
+        }
+      );
+
+      expect(mockUseIssueDetails.dispatch).toHaveBeenCalledWith({
+        type: 'UPDATE_EVENT_SECTION',
+        key: 'highlights',
+        config: {initialCollapse: false}, // Should override initialCollapse when preventCollapse is true
+      });
+    });
+  });
+
+  describe('Error boundary', () => {
+    it('wraps content in error boundary', () => {
+      const ThrowError = () => {
+        throw new Error('Test error');
+      };
+
+      // Suppress console.error for this test
+      const consoleSpy = jest.spyOn(console, 'error').mockImplementation();
+
+      render(
+        <FoldSection title="Test Section" sectionKey={SectionKey.HIGHLIGHTS}>
+          <ThrowError />
+        </FoldSection>
+      );
+
+      expect(screen.queryByText('Test error')).not.toBeInTheDocument();
+      consoleSpy.mockRestore();
+    });
+  });
+
+  describe('Custom props', () => {
+    it('applies className prop', () => {
+      render(
+        <FoldSection
+          title="Test Section"
+          sectionKey={SectionKey.HIGHLIGHTS}
+          className="custom-class"
+        >
+          <div>Test Content</div>
+        </FoldSection>
+      );
+
+      expect(screen.getByRole('region')).toHaveClass('custom-class');
+    });
+
+    it('applies style prop', () => {
+      render(
+        <FoldSection
+          title="Test Section"
+          sectionKey={SectionKey.HIGHLIGHTS}
+          style={{backgroundColor: 'red'}}
+        >
+          <div>Test Content</div>
+        </FoldSection>
+      );
+
+      // Styles are applied through CSS-in-JS, just verify the prop is accepted
+      const section = screen.getByRole('region');
+      expect(section).toBeInTheDocument();
+    });
+
+    it('applies custom data-test-id', () => {
+      render(
+        <FoldSection
+          title="Test Section"
+          sectionKey={SectionKey.HIGHLIGHTS}
+          dataTestId="custom-test-id"
+        >
+          <div>Test Content</div>
+        </FoldSection>
+      );
+
+      expect(screen.getByTestId('custom-test-id')).toBeInTheDocument();
+    });
+
+    it('uses default data-test-id when not provided', () => {
+      render(
+        <FoldSection title="Test Section" sectionKey={SectionKey.HIGHLIGHTS}>
+          <div>Test Content</div>
+        </FoldSection>
+      );
+
+      expect(screen.getByTestId('highlights')).toBeInTheDocument();
+    });
```

Comment:
delete

Reply:
Same comment as above, let me just finish making the port and I'll delete it. I'm going to stack a PR on top of this one so this won't get merged